### PR TITLE
feat(terraform): add CKV NCP rules about server instance and public IP

### PR DESCRIPTION
**Pull request by pj991207**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description
Add terraform scan rules about provider Naver Cloud Platform.
CKV_NCP_23 is associated with "ncloud server instance should not have Public IP"

### Description
https://registry.terraform.io/providers/NaverCloudPlatform/ncloud/latest/docs/resources/public_ip

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nimrodkor — APPROVED** [review]

File: checkov/terraform/checks/resource/ncp/ServerPublicIP.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.terraform.checks.resource.base_resource_check import BaseResourceCheck
+
+
+class ServerPublicIP(BaseResourceCheck):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/ncp/ServerPublicIP.py:5]: you can change this to leverage `BaseResourceNegativeValueCheck` like here https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/resource/aws/IAMPolicyAttachedToGroupOrRoles.py 
the `get_inspected_key` would be `server_instance_no` for your check.

Comment (pj991207) [reply checkov/terraform/checks/resource/ncp/ServerPublicIP.py]: Thanks. I reflect your advice

**Review by nimrodkor — APPROVED** [review]
TY!

**Review by gruebel — APPROVED** [review]
🍺

**Pull request merged**